kitsune/messages/utils.py

```diff
@@ -18,38 +16,52 @@ def send_message(to, text=None, sender=None):
     # We need a sender, a message, and someone to send it to
     if not sender or not text or not to:
         return
-    # This is users from the To field
-    users = {user for user in to if isinstance(user, User)}
-    # This is groups from the To field
-    groups = {group for group in to if isinstance(group, Group)}
-    # This is all the users that are going to receive the message
-    receivers = users | set(User.objects.filter(groups__in=groups).distinct())
+    # User pks from users in To field
+    users = []
+    groups = []
+
+    for obj in to:
+        if isinstance(obj, User):
+            users.append(obj)
+        else:
+            groups.append(obj)
+
+    # Resolve all unique user ids, including those in groups
+    # We need to keep group_users separate for email notifications
+    all_recipients_of_message = set(user.id for user in users) | set(
+        User.objects.filter(groups__in=groups).values_list("id", flat=True)
+    )
 
+    # Create the outbox message
     outbox_message = OutboxMessage.objects.create(sender=sender, message=text)
-    # Add the users from the To field to the outbox message, not all the users
-    # that are going to receive the message - this way we don't overwhelm the
-    # message UI
     outbox_message.to.set(users)
+    outbox_message.to_group.set(groups)
 
-    if groups:
-        # Add the groups from the To field to the message
-        outbox_message.to_group.set(groups)
-
-    set_of_user_pks_to_email_private_message = set(
+    # Fetch settings for email notifications in one go
+    # For now, we only send emails to users who were individually
+    # listed in 'to' field, not to users who are from groups
+    users_to_email = set(
         Setting.objects.filter(
-            user__in=receivers, name="email_private_messages", value=True
-        ).values_list("user__pk", flat=True)
+            user__in=users, name="email_private_messages", value=True
```

Comment:
Just wanted to record that I'd prefer this to use `all_recipients_of_message` instead of just the `users`, but that's for another discussion.

Reply:
Got  it.